sql-statements/sql-statement-commit.md

```diff
@@ -38,8 +38,8 @@ Query OK, 0 rows affected (0.01 sec)
 
 ## MySQL compatibility
 
-* In MySQL, with the exception of Group Replication with multiple primaries, it is not typical that a `COMMIT` statement could result in an error. By contrast, TiDB uses optimistic concurrency control and conflicts may result in `COMMIT` returning an error.
-* Be default, `UNIQUE` and `PRIMARY KEY` constraint checks are deferred until statement commit. This behavior can be changed by setting `tidb_constraint_check_in_place=TRUE`.
+* By default, `UNIQUE` and `PRIMARY KEY` constraint checks are deferred until statement commit. This improves performance because checks are batched, but may result in compatibility issues for applications not expecting a commit statement to fail. This behavior can be changed by setting `tidb_constraint_check_in_place=TRUE`.
```

Comment:
I'm pretty sure constraint checks are deferred only in optimistic transaction/locking mode. This should be made more clear throughout the documentation.

```
mysql> select @@tidb_constraint_check_in_place;
+----------------------------------+
| @@tidb_constraint_check_in_place |
+----------------------------------+
| 0                                |
+----------------------------------+
1 row in set (0.00 sec)

mysql> begin pessimistic;
Query OK, 0 rows affected (0.00 sec)

mysql> INSERT INTO users (username) VALUES ('jane'), ('chris'), ('bill');
ERROR 1062 (23000): Duplicate entry 'bill' for key 'username'
mysql> rollback;
Query OK, 0 rows affected (0.00 sec)

mysql> begin optimistic;
Query OK, 0 rows affected (0.00 sec)

mysql> INSERT INTO users (username) VALUES ('jane'), ('chris'), ('bill');
Query OK, 3 rows affected (0.00 sec)
Records: 3  Duplicates: 0  Warnings: 0

mysql> commit;
ERROR 1062 (23000): Duplicate entry 'bill' for key 'username'
```

Reply:
Thanks, there are a few places that will need editing.